qiskit/tools/qcvv/tomography.py

```diff
@@ -36,295 +35,592 @@
-    build_process_tomography_circuits function.
+    The quantum program must contain a circuit 'name', which is treated as a
+    state preparation circuit for state tomography, or as teh circuit being
+    measured for process tomography. This function then appends the circuit
+    with a set of measurements specified by the input `tomography_set`,
+    optionally it also prepends the circuit with state preparation circuits if
+    they are specified in the `tomography_set`.
+
+    For n-qubit tomography with a tomographically complete set of preparations
+    and measurements this results in $4^n 3^n$ circuits being added to the
+    quantum program.
 
     Args:
-        name (string): the name of the original circuit to be
-                       reconstructed.
-        qubits: (list[int]): the qubits being measured.
+        qp (QuantumProgram): A quantum program to store the circuits.
+        name (string): The name of the base circuit to be appended.
+        qubits (list[int]): a list of the qubit indexes of qreg to be measured.
+        qreg (QuantumRegister): the quantum register containing qubits to be
+                                measured.
+        creg (ClassicalRegister): the classical register containing bits to
+                                  store measurement outcomes.
+        tomoset (tomography_set): the dict of tomography configurations.
+        silent (bool, optional): hide verbose output.
 
     Returns:
-        A list of circuit names.
+        A list of names of the added quantum state tomography circuits.
+
+    Example:
+        For a tomography set  specififying state tomography of qubit-0 prepared
+        by a circuit 'circ' this would return:
+        ```
+        ['circ_meas_X(0)', 'circ_meas_Y(0)', 'circ_meas_Z(0)']
+        ```
+        For process tomography of the same circuit with preparation in the
+        SIC-POVM basis it would return:
+        ```
+        [
+            'circ_prep_S0(0)_meas_X(0)', 'circ_prep_S0(0)_meas_Y(0)',
+            'circ_prep_S0(0)_meas_Z(0)', 'circ_prep_S1(0)_meas_X(0)',
+            'circ_prep_S1(0)_meas_Y(0)', 'circ_prep_S1(0)_meas_Z(0)',
+            'circ_prep_S2(0)_meas_X(0)', 'circ_prep_S2(0)_meas_Y(0)',
+            'circ_prep_S2(0)_meas_Z(0)', 'circ_prep_S3(0)_meas_X(0)',
+            'circ_prep_S3(0)_meas_Y(0)', 'circ_prep_S3(0)_meas_Z(0)'
+        ]
+        ```
     """
-    preps = __tomo_labels(name + '_prep', qubits, states=True)
-    return reduce(lambda acc, c:
-                  acc + __tomo_labels(c + '_meas', qubits),
-                  preps, [])
 
+    dics = tomoset['circuits']
+    labels = tomography_circuit_names(tomoset, name)
+    circuit = qp.get_circuit(name)
+
+    for label, conf in zip(labels, dics):
+        tmp = circuit
+        # Add prep circuits
+        if 'prep' in conf:
+            prep = qp.create_circuit('tmp_prep', [qreg], [creg])
+            for q, op in conf['prep'].items():
```

Comment:
`q` -> ?
Does `op` mean `operation` ? (which is a widely accepted contraction) ... otherwise, could it be renamed too?